src/decorators.ts

```diff
@@ -48,31 +48,30 @@ export interface PropertyOptions {
 
 function createProperty(
     proto: any, name: string, options?: PropertyOptions): void {
-  const notify = options && options.notify || false;
-  const reflectToAttribute = options && options.reflectToAttribute || false;
-  const readOnly = options && options.readOnly || false;
-  const computed = options && options.computed || '';
-  const observer = options && options.observer || '';
-
-  let type;
-  if (options && options.hasOwnProperty('type')) {
-    type = options.type;
-  } else if (
-      (window as any).Reflect && Reflect.hasMetadata && Reflect.getMetadata &&
-      Reflect.hasMetadata('design:type', proto, name)) {
-    type = Reflect.getMetadata('design:type', proto, name);
-  } else {
-    console.error(
-        `A type could not be found for ${name}. ` +
-        'Set a type or configure Metadata Reflection API support.');
-  }
-
   if (!proto.constructor.hasOwnProperty('properties')) {
     Object.defineProperty(proto.constructor, 'properties', {value: {}});
   }
 
-  const finalOpts: PropertyOptions =
-      {type, notify, reflectToAttribute, readOnly, computed, observer};
+  let finalOpts: PropertyOptions =
```

Comment:
`const mergedOpts: PropertyOptions = {proto.constructor.properties[name] || {}, ...options};`?

Reply:
Eh nevermind, forgot that options can be undefined.

Reply:
well it looked a little nasty. because `options` is optional too...

so its `proto.constructor.properties[name] || {}` merged with `options || {}`

Reply:
ah nevermind i did not realise spread works fine with `undefined`!

Reply:
Cool! So concise now.